### ISSUE TITLE
Animation: swapParent

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#2528]
## Steps to Reproduce

```
- a
  - b
```

1. Set the cursor on `b`
2. Swap Parent

## Current Behavior

No animation.

https://github.com/user-attachments/assets/cdf4b34a-a3fc-4083-9deb-b5f6e1dbd0ce

## Expected Behavior

- [x] `a` should curve left and `b` should curve right (counter-clockwise) as they animate into their new positions.
- [x] The greater thought (via https://github.com/cybersemics/em/blob/main/src/util/compareThought.ts#L188) should curve right, while the lesser thought should curve left.

## Notes

I recommend [layered animation](https://tobiasahlin.com/blog/curved-path-animations-in-css/) to simulate animating along a curve.

**Comment by zukilover** [comment]
Currently, swapping the parent does not trigger position changes. I'm still working on finding a way to replicate the `moveThought` functionality and integrate it into the swapParent action.

**Comment by raineorshine** [comment]
Yes, that's tricky. I wonder if you could alter the `key` to trick React into swapping the parent and child, thus causing them to have an animated position change.

**Comment by zukilover** [comment]
I added a `value` to the key, which isn't the ideal solution, but I did it just to test if it would trigger the animation. The result was that it triggered the thought fade transition, but not the x and y transitions for the tree nodes. I’m considering an approach where we first swap the nodes, then only swap the thought IDs (not the values) to trigger the x and y transitions. I’m open to any suggestions or alternative ideas.

**Comment by raineorshine** [comment]
I'm not sure off the top of my head. Since the ids and thus keys are "wrong" due to the `swapParent` logic, it seems like we will need to swap DOM nodes with no animation either at the beginning or end of the swap in order. Then the animation will at least start or end with the correct key.
